src/services/edusharing/logic/connector.js

```diff
@@ -0,0 +1,270 @@
+			console.error('Oauth failed');
+			return null;
+		});
+	}
+
+	allConfigurationValuesHaveBeenDefined() {
+		return (
+			Configuration.get('ES_DOMAIN')
+			&& Configuration.get('ES_USER')
+			&& Configuration.get('ES_PASSWORD')
+			&& Configuration.get('ES_GRANT_TYPE')
+			&& Configuration.get('ES_OAUTH_SECRET')
+			&& Configuration.get('ES_CLIENT_ID')
+		);
+	}
+
+	async login() {
+		this.authorization = await this.getCookie();
+		this.accessToken = await this.getAuth();
+	}
+
+	isLoggedin() {
+		// returns false if cookie or accesstoken is falsy
+		return !!this.authorization && !!this.accessToken;
+	}
+
+	async GET(id, params) {
+		if (!this.allConfigurationValuesHaveBeenDefined()) {
+			throw configurationIncompleteError();
+		}
+
+		if (this.isLoggedin() === false) {
+			await this.login();
+		}
+
+		const propertyFilter = '-all-';
+
+		const options = {
+			method: 'GET',
+			// eslint-disable-next-line max-len
+			url: `${Configuration.get('ES_DOMAIN')
+			}/edu-sharing/rest/node/v1/nodes/mv-repo.schul-cloud.org/${id
+			}/metadata?propertyFilter=${propertyFilter}`,
+			headers: {
+				...EduSharingConnector.headers,
+				cookie: this.authorization,
+			},
+			timeout: REQUEST_TIMEOUT,
+		};
+
+		const eduResponse = await requestRepeater(options);
+		return eduResponse.node;
+	}
+
+	async FIND(data) {
+		const contentType = data.query.contentType || 'FILES'; // enum:[FILES,FILES_AND_FOLDERS,COLLECTIONS,ALL]
+		const skipCount = parseInt(data.query.$skip, 10) || 0;
+		const maxItems = parseInt(data.query.$limit, 10) || 9;
+		const sortProperties = data.query.sortProperties || 'score';
+		const sortAscending = false;
+		const propertyFilter = '-all-'; // '-all-' for all properties OR ccm-stuff
+		const searchWord = data.query.searchQuery || '';
+
+		if (!this.allConfigurationValuesHaveBeenDefined()) {
+			throw configurationIncompleteError();
+		}
+
+		if (this.isLoggedin() === false) {
+			await this.login();
+		}
+
+		const urlBase = `${Configuration.get('ES_DOMAIN')
+		}/edu-sharing/rest/search/v1/queriesV2/mv-repo.schul-cloud.org/mds/ngsearch/?`;
+		const url = urlBase
+			+ [
+				`contentType=${contentType}`,
+				`skipCount=${skipCount}`,
+				`maxItems=${maxItems}`,
+				`sortProperties=${sortProperties}`,
+				`sortAscending=${sortAscending}`,
+				`propertyFilter=${propertyFilter}`,
+			].join('&');
+
+		const options = {
+			method: 'POST',
+			// This will be changed later with a qs where sorting, filtering etc is present.
+			// eslint-disable-next-line max-len
+			url,
+			headers: {
+				...EduSharingConnector.headers,
+				cookie: this.authorization,
+			},
+			body: JSON.stringify({
+				criterias: [
+					{ property: 'ngsearchword', values: [`${searchWord}`] },
+				],
+				facettes: ['cclom:general_keyword'],
+			}),
+			timeout: REQUEST_TIMEOUT,
+		};
+
+		const parsed = await requestRepeater(options);
+
+		// provided by client eg data.query.filterOptions
+		const filterOptions = {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['filterOptions' is assigned a value but never used.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=4977295)